monitoring/daily_report.py

```
@@ -0,0 +1,654 @@
+        stats['error'] = 'Log file not found'
+        return stats
+
+    # Use set to count unique visitors (hashed IPs, discarded after counting)
+    unique_visitor_hashes: Set[int] = set()
+
+    target_date_str = target_date.strftime('%Y-%m-%d')
+
+    try:
+        # Try reading as plain text first, then gzipped
+        if log_path.suffix == '.gz':
+            open_func = lambda p: gzip.open(p, 'rt', encoding='utf-8')
```

Ruff (0.14.14)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `open_func` as a `def`

(E731)